CMakeLists.txt

```diff
@@ -0,0 +1,207 @@
+option(PCAPPP_BUILD_EXAMPLES "Build Examples" ${PCAPPP_MAIN_PROJECT})
+cmake_dependent_option(
+  PCAPPP_BUILD_TUTORIALS
+  "Build Tutorials"
+  OFF
+  "PCAPPP_BUILD_EXAMPLES"
+  OFF)
+option(PCAPPP_BUILD_TESTS "Build Tests" ${PCAPPP_MAIN_PROJECT})
+option(BUILD_SHARED_LIBS "Build using shared libraries" OFF)
+
+option(PCAPPP_USE_DPDK "Setup PcapPlusPlus with DPDK. In this case you must also set DPDK_HOME")
+option(PCAPPP_USE_PF_RING "Setup PcapPlusPlus with PF_RING. In this case you must also set PF_RING_ROOT")
+option(PCAPPP_INSTALL "Install Pcap++" ${PCAPPP_MAIN_PROJECT})
+
+# Set C++11
+set(CMAKE_CXX_STANDARD 11)
+# popen()/pclose() are not C++ standards
+set(CMAKE_CXX_EXTENSIONS ON)
+
+# Usually on Windows PCAP_ROOT and Packet_ROOT are at the same location
+if(PCAPPP_TARGET_PLATFORM_WINDOWS)
+  if(NOT PCAP_ROOT)
+    message(FATAL_ERROR "Please specify Npcap/WinPcap SDK directory with -DPCAP_ROOT=<PCAP_SDK_PATH>")
+  endif()
+  if(NOT Packet_ROOT)
+    set(Packet_ROOT ${PCAP_ROOT})
+  endif()
+endif()
+
+# Do we need Packet Packages on Windows ?
+if(PCAPPP_TARGET_PLATFORM_WINDOWS)
+  find_package(Packet)
+  if(NOT Packet_FOUND)
+    message(FATAL_ERROR "Please specify Packet library -DPacket_ROOT=<NPCAP_SDK_PATH>")
+  endif()
+endif()
+
+find_package(PCAP)
+if(NOT PCAP_FOUND)
+  if(PCAPPP_TARGET_PLATFORM_WINDOWS)
+    message(FATAL_ERROR "Please specify Npcap/WinPcap SDK directory with -DPCAP_ROOT=<PCAP_SDK_PATH>")
+  else()
+    message(FATAL_ERROR "PCAP library not found!")
+  endif()
+endif()
+
+# Enable the option if it's available
+option(PCAPPP_HAS_PCAP_IMMEDIATE_MODE "Enable PCAP immediate Mode (supported on libpcap>=1.5)"
+       ${HAVE_PCAP_IMMEDIATE_MODE})
+
+# Check in case user force it but it's not available
+if(PCAPPP_HAS_PCAP_IMMEDIATE_MODE)
+  if(NOT HAVE_PCAP_IMMEDIATE_MODE)
+    message(FATAL_ERROR "PCAP library doesn't have Immediate Mode support!")
+  endif()
+  add_definitions(-DHAS_PCAP_IMMEDIATE_MODE)
+endif()
+
+# Enable the option if it's available
+option(PCAPPP_HAS_PCAP_SET_DIRECTION
+       "Enable set direction for capturing incoming or outgoing packets (supported on libpcap>=0.9.1)" OFF)
+
+# Check in case user force it but it's not available
+if(PCAPPP_HAS_PCAP_SET_DIRECTION)
+  if(NOT HAVE_PCAP_DIRECTION)
+    message(FATAL_ERROR "PCAP library doesn't have Direction support!")
+  endif()
+  add_definitions(-DHAS_SET_DIRECTION_ENABLED)
+endif()
+
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+if(PCAPPP_TARGET_PLATFORM_WINDOWS AND PCAPPP_TARGET_COMPILER_MSVC)
+  # Other hacks for VStudio
+  add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+else()
+  if(NOT CMAKE_USE_PTHREADS_INIT)
+    message(FATAL_ERROR "Pthreads not found!")
+  endif()
+endif()
+
+if(PCAPPP_USE_DPDK)
+  find_package(DPDK)
+  if(NOT dpdk_FOUND)
+    message(FATAL_ERROR "Specify DPDK Library with env DPDK_DIR=<DPDK_PATH>")
+  endif()
+  add_definitions(-DUSE_DPDK)
+endif()
```

Comment:
An important part we missed is these lines in `configure-linux.sh`:
https://github.com/seladb/PcapPlusPlus/blob/master/configure-linux.sh#L327-L333

There is a DPDK setup script written in python that should be run before a running a DPDK app. It sets up hugepages, the DPDK driver, and moves control over the NIC from the OS to DPDK.

This file is called `setup_dpdk.py` and resides in `mk/setup_dpdk.py`.

During the configuration process we copy it to the root directory of PcapPlusPlus and also create a file called `setup_dpdk_settings.dat` which contains the DPDK directory. This file is needed by `setup_dpdk.py` to save its state. 

Is there a way to do all that in CMake?

Reply:
yes we can, would be nice if we could pass the RTE_SDK by arg instead of having a temporary .dat file

Reply:
something like this should work: 
https://github.com/seladb/PcapPlusPlus/pull/944/commits/c41aac00aba74e0cdaa3be6848ac0a2897946eee

Reply:
I don't think CMake should run this script, it should just copy it somewhere that will be clear to users, and then create the `setup_dpdk_settings.dat` file.
This script should be run regardless of the build, for example: after every restart

Reply:
Ok it's kind of custom helper generated by Pcap++ to help using DPDK.

What about installing it ?

Reply:
I don't know if installing it is the right solution here...
DPDK has its own tools inside the repo: https://github.com/DPDK/dpdk/tree/main/usertools

Maybe we should eventually move this script to a similar folder (like `devtools` or `usertools`)?

For now I suggest we do the same as the current build system - copy it from `mk/` to the root directory and create the `setup_dpdk_settings.dat`. Later, when we remove the `mk/` directory we can decide what to do with this script. What do you think?

Reply:
these tools are installed under /usr/local/share/dpdk/usertools/* 

Will be nice to install it their too no?

Reply:
I don't think PcapPlusPlus should install DPDK tools...

Reply:
Yep, i have added an example unfortunately CMake doesn't where are the DPDK source, it will look for include and libraries but they can be installed in /usr/local/. So it will not give the DPDK source folder

Reply:
Maybe we can do what I suggested here: https://github.com/seladb/PcapPlusPlus/pull/944#discussion_r963390606 ?

That can make everything simpler. I can do this change in a separate PR that we can merge before this PR. Let me know what you think

Reply:
I'm resolving this comment and keeping https://github.com/seladb/PcapPlusPlus/pull/944#discussion_r963390606